contracts/staking/PoSValidatorManager.sol

```diff
@@ -129,25 +139,44 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         uint32 messageIndex
     ) external {
         PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
+
+        Validator memory validator = _initializeEndValidation(validationID);
+
         // Check that minimum stake duration has passed
-        Validator memory validator = getValidator(validationID);
         require(
-            block.timestamp
+            validator.endedAt
                 >= validator.startedAt + $._validatorRequirements[validationID].minStakeDuration,
             "PoSValidatorManager: minimum stake duration not met"
         );
 
         if (includeUptimeProof) {
-            _getUptime(validationID, messageIndex);
+            uint64 uptime = _getUptime(validationID, messageIndex);
+            if (
+                uptime * 100
+                    >= (validator.endedAt - validator.startedAt) * UPTIME_REWARDS_THRESHOLD_PERCENTAGE
+            ) {
+                $._redeemableValidatorRewards[validator.owner] += $
```

Comment:
[Not for this PR - can be after the initial audit]

Primary network validations allow for specifying a different rewards address than the address that created the validation. Would be good to support that in the contracts as well.

Reply:
https://github.com/ava-labs/teleporter/issues/555